spec/app/domain/util/socket_server_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Util::SocketService do
+
+  let(:service)  do
+    Util::SocketService.new(
+      socket: socket
+    )
+  end
+
+  describe('#run') do
+    context 'When a socket is started and the socket file already exists' do
+      let(:socket) {'/run/authn-local/.socket'}
+      it 'raises and error' do
+        expect do
+          service.run do |passed_arguments|
+            Authentication::AuthnOidc::V2::Commands::ListProviders.new.call(
+              message: passed_arguments
+            )
+          end
+        end.to raise_error("Socket: /run/authn-local/.socket already exists")
+      end
+    end
+
+    context 'When a socket is started and directory ca not be found' do
+      let(:socket) {'/run/test/.socket'}
+      it 'raises and error' do
+        expect do
+          service.run do |passed_arguments|
+            Authentication::AuthnOidc::V2::Commands::ListProviders.new.call(
+              message: passed_arguments
+            )
+          end
+        end.to raise_error('Socket Service requires directory "/run/test" to exist and be a directory')
+      end
+    end
+
+    context 'When a socket is started and given a code block' do
+      let(:socket) {'/run/authn-local/.socket100'}
+      it 'will handle the input with the code block' do
+        thread = Thread.new do
+          service.run do |passed_arguments|
+             passed_arguments
+          end
+        end
+        thread.join(10)
```

Comment:
I ran into problems running this locally (server was running in the foreground.  I solved it with the following:
```suggestion
        # Allow the socket server time to start
        sleep(2)
```
`join` holds the thread open (for ten seconds in this case), which isn't what we want.  We need the server in the background so RSpec can run in the foreground.

Reply:
it works locally for me, but doesnt join hold it open until its complete or it hits 10? i just didnt want to do a sleep cuz it might get flakey depending on how long it takes for the socket to open. and i had it failing sometimes with sleep 5